### ISSUE TITLE
Replace pyarlo in HA (https://github.com/tchellomello/python-arlo)?

### ISSUE THREAD

**Issue by ryanwinter** [twrecked/hass-aarlo#71]
Not really an issue, but unsure where else to put this.

Is this library likely to replace the pyarlo in HA? That library seems largely abandoned and doesn't support newer cameras like the Arlo Q very well.

I plan to give this a whirl later this week, mostly looking for Arlo Q support, particularly motion detection events and ability to arm/disarm.

**Comment by twrecked** [comment]
The plan is to provide a new module in the HA so people happy with the old module can keep using it. 

It's taken a little longer to get there than I wanted because spring arrived and my real life got in the way. And opening the code up to a wider audience meant a steady supply of bugs and unknowns to deal with.

I'm planning to get one final release out this week (0.5) and then I'll work on tidying the code and try getting it into HA.

And give it a go, let me know how it works out!

**Comment by ryanwinter** [comment]
Sounds great! I'll give it a whirl and see how it works with the Arlo Q cameras. I noticed that the arm/disarm code seems to be stubbed out if there is no basestation, so would be interested to see if this needs to be expanded.

**Comment by twrecked** [comment]
Which code is stubbed out?

And your Arlo Q cameras should appear as cameras and base stations. (And binary sensors if you enable that too.)

**Comment by ryanwinter** [comment]
Ah interesting, I was looking at this code:
```
    def set_base_station_mode(self, mode):
        """Set the mode in the base station."""
        # Get the list of base stations identified by library
        base_stations = self.hass.data[DATA_ARLO].base_stations

        # Some Arlo cameras does not have base station
        # So check if there is base station detected first
        # if yes, then choose the primary base station
        # Set the mode on the chosen base station
        if base_stations:
            primary_base_station = base_stations[0]
            primary_base_station.mode = mode
```

It grabs a list of base stations from hass, and then sets the mode on the first one. Now I have 5 Arlo Q cameras, so you are saying that I will then have 5 base stations (if each camera is also a base station)? But the code always seems to grab the first base station.

**Comment by twrecked** [comment]
I don't think that code gets called, I probably left it in from the original pyarlo code. I fully intend to write some code to enable/disable motion detection but was going to base it on the Smartthings model - create a custom mode you change rather than arming and disarming the entire base station.

Alarm arm/disarm will come through the alarm_control_panel code.

**Comment by ryanwinter** [comment]
Yeah, so I realise now that the actual code is in the pyaarlo directory :)

**Comment by twrecked** [comment]
Some of it :) There are 2 pieces:

`aarlo/pyaarlo` This is the low level code
`aarlo` This is the home-assistant level code

My plan is to eventually split them and make the `pyaarlo` a pip component and a requirement for `aarlo`. I've not done that during this initial phase to make make the development cycle a little easier.

**Comment by ryanwinter** [comment]
It's confused by the fact that the ArloQ are all individual camera's, which you can easily arm and disarm separately, compared to the battery arlos, where there is a single mode for all the camera's.

The standard scenario, is that I would want inside cameras to be disarmed at certain times, while the outside ones continue to operate normally.

1. With Arlo Q/Baby, you simple call disarm on both the internal cameras individually
2. With Arlo Pro/Ultra, you need to create a custom mode, and set the inside cameras to inactive, and the outside cameras to active.

At the moment, I just leave all my devices armed at all times, and then I just use HA to throw away alert messages for the inside devices when I'm home.

**Comment by twrecked** [comment]
Full disclaimer: I don't have any ArloQ cameras only battery powered ones.

I think I'm misunderstanding you, but why wouldn't you call:
 
```
service: alarm_control_panel.alarm_disarm
 -or-
service: alarm_control_panel.alarm_arm_away
 -or-
service: alarm_control_panel.alarm_arm_home
with:
{
  "entity_id": "alarm_control_panel.aarlo_your_base"
}
```
(Again, I don't have Q cameras!)

**Comment by ryanwinter** [comment]
Disclaimer from me, I don't own any battery operated ones :)

ArloQ cameras don't have a concept of alarm control panel or base station, this is what my arlo mode page looks like.

![image](https://user-images.githubusercontent.com/10238039/59381585-99276600-8d10-11e9-9d33-8f08457e2adc.png)

So what would make more sense, is to the service camera.enable_motion_detection or camera.disable_motion_detection which would simply arm/disarm the camera.

**Comment by twrecked** [comment]
I get you. I've just never got round to fixing that. You ok trying a branch?

**Comment by twrecked** [comment]
https://github.com/twrecked/hass-aarlo/tree/arloq-motion

This might fix the issue. Or make the feature work!

**Comment by ryanwinter** [comment]
Tried it, and it works perfectly! Both the motion binary sensor (can get rid of the crappy IFTTT applet), and I was able to arm/disarm the devices perfectly.

Also the camera card is really good!

**Comment by twrecked** [comment]
That's great news. I merged the code into the master branch with #72
